### ISSUE TITLE
Ability to perceive chance factor as probability.

### ISSUE THREAD

**Issue by barannnnnns** [slvlr/PM2-Plugins#59]
Plugin: Advanced Crates
Minecraft: 1.12.1
  Paper.

Chance is a tendency (mathematical).
So when we say chance: 100, this product will not be given with certainty when we open the crate. but if we take it as a probability, this product will be given for sure when we open the crate.

I want to use it as a probability because the tendency (i.e. the current version of the plugin) happens randomly in crates with too many items. It doesn't matter if I write chance:100 or chance:1.
I think this is a very important situation.

set-chance-as-probability: true  # this option will treat as probability where it says chance.

I can't adjust item rarity in multi-reward crates in the current version of the plugin.  Please review this situation yourself.

**Comment by ThePM2** [comment]
you see there's an issue with this type of probability configuration, as in how would the rewards chance be prioritized. The only proper way with the requested chance method is having the rewards go 1:1 in having a proc to win based on the chance.

If you roll for these rewards one by one, starting with the first reward, the actual probabilities for each reward would be affected by whether the preceding rewards are won or not.

For example, if you win Reward 1 (5% chance), then you won't even get to the roll for Reward 2 or Reward 3. This means that the 90% and 50% probabilities for Rewards 2 and 3 are not "pure" probabilities in this sequential rolling system, because their chances are actually conditional on not winning any of the earlier rewards.

or in simpler terms:
If you roll for these rewards one by one, starting with the first reward, the actual probabilities for each reward would be affected by whether the preceding rewards are won or not.

For example, if you win Reward 1 (5% chance), then you won't even get to the roll for Reward 2 or Reward 3. This means that the 90% and 50% probabilities for Rewards 2 and 3 are not "pure" probabilities in this sequential rolling system, because their chances are actually conditional on not winning any of the earlier rewards.

**Comment by barannnnnns** [comment]
I understand what you mean. Thank you for this descriptive answer.
